api/background/handlers/ethereum/accounts.ts

```diff
@@ -0,0 +1,33 @@
+import { RpcError, ApiRequestWithHost } from "@/api/message";
+import { ApiUtils } from "@/api/background/utils";
+import { publicKeyToAddress } from "viem/accounts";
+import { uncompressPublicKey } from "./utils";
+
+export const accountsHandler = async (
+  tabId: number,
+  message: ApiRequestWithHost,
+  sendResponse: (response?: any) => void,
+) => {
+  const sendError = function (error: RpcError) {
+    // NOTE: can not send undefined as response, so send null
+    sendResponse(ApiUtils.createApiResponse(message.id, null, error));
+  };
+
+  const isConnected = await ApiUtils.isHostConnected(message.host);
+  const isUnlocked = ApiUtils.isUnlocked();
+  if (!isConnected || !isUnlocked) {
+    sendResponse(ApiUtils.createApiResponse(message.id, []));
+    return;
+  }
+
+  const account = await ApiUtils.getCurrentAccount();
+  if (!account?.publicKeys) {
```

Comment:
maybe check also for the `length !== 0` since you are grabbing the first element right after

Reply:
https://github.com/forbole/kastle/pull/90/commits/570ddf69f15db588a2bdd2be2581973612adace0